File: src/main/java/model/Coupon.java
package model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import org.hibernate.annotations.ColumnDefault;
import org.hibernate.annotations.Nationalized;

import java.math.BigDecimal;
import java.time.Instant;

@Entity
@Table(name = "Coupons")
public class Coupon {
    @Id
    @Column(name = "CouponID", nullable = false)
    private Integer id;

    @Nationalized
    @Column(name = "CouponCode", nullable = false, length = 50)
    private String couponCode;

    @Nationalized
    @Column(name = "CouponName", nullable = false)
    private String couponName;

    @Nationalized
    @Column(name = "Description", length = 500)
    private String description;

    @Nationalized
    @Column(name = "DiscountType", nullable = false, length = 20)
    private String discountType;

    @Column(name = "DiscountValue", nullable = false, precision = 18, scale = 2)
    private BigDecimal discountValue;

    @ColumnDefault("0")
    @Column(name = "MinOrderAmount", precision = 18, scale = 2)
    private BigDecimal minOrderAmount;

    @Column(name = "MaxDiscountAmount", precision = 18, scale = 2)
    private BigDecimal maxDiscountAmount;

    @Column(name = "UsageLimit")
    private Integer usageLimit;

    @ColumnDefault("0")
    @Column(name = "UsedCount")
    private Integer usedCount;

    @ColumnDefault("1")
    @Column(name = "IsActive")
    private Boolean isActive;

    @Column(name = "StartDate", nullable = false)
    private Instant startDate;

    @Column(name = "EndDate", nullable = false)
    private Instant endDate;

    @ColumnDefault("getdate()")
    @Column(name = "CreatedDate")
    private Instant createdDate;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getCouponCode() {
        return couponCode;
    }

    public void setCouponCode(String couponCode) {
        this.couponCode = couponCode;
    }

    public String getCouponName() {
        return couponName;
    }

    public void setCouponName(String couponName) {
        this.couponName = couponName;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getDiscountType() {
        return discountType;
    }

    public void setDiscountType(String discountType) {
        this.discountType = discountType;
    }

    public BigDecimal getDiscountValue() {
        return discountValue;
    }

    public void setDiscountValue(BigDecimal discountValue) {
        this.discountValue = discountValue;
    }

    public BigDecimal getMinOrderAmount() {
        return minOrderAmount;
    }

    public void setMinOrderAmount(BigDecimal minOrderAmount) {
        this.minOrderAmount = minOrderAmount;
    }

    public BigDecimal getMaxDiscountAmount() {
        return maxDiscountAmount;
    }

    public void setMaxDiscountAmount(BigDecimal maxDiscountAmount) {
        this.maxDiscountAmount = maxDiscountAmount;
    }

    public Integer getUsageLimit() {
        return usageLimit;
    }

    public void setUsageLimit(Integer usageLimit) {
        this.usageLimit = usageLimit;
    }

    public Integer getUsedCount() {
        return usedCount;
    }

    public void setUsedCount(Integer usedCount) {
        this.usedCount = usedCount;
    }

    public Boolean getIsActive() {
        return isActive;
    }

    public void setIsActive(Boolean isActive) {
        this.isActive = isActive;
    }

    public Instant getStartDate() {
        return startDate;
    }

    public void setStartDate(Instant startDate) {
        this.startDate = startDate;
    }

    public Instant getEndDate() {
        return endDate;
    }

    public void setEndDate(Instant endDate) {
        this.endDate = endDate;
    }

    public Instant getCreatedDate() {
        return createdDate;
    }

    public void setCreatedDate(Instant createdDate) {
        this.createdDate = createdDate;
    }

}

File: src/main/java/model/Coupon.kt
package model

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import org.hibernate.annotations.ColumnDefault
import org.hibernate.annotations.Nationalized
import java.math.BigDecimal
import java.time.Instant

@Entity
@Table(name = "Coupons")
open class Coupon {
    @Id
    @Column(name = "CouponID", nullable = false)
    open var id: Int? = null

    @Nationalized
    @Column(name = "CouponCode", nullable = false, length = 50)
    open var couponCode: String? = null

    @Nationalized
    @Column(name = "CouponName", nullable = false)
    open var couponName: String? = null

    @Nationalized
    @Column(name = "Description", length = 500)
    open var description: String? = null

    @Nationalized
    @Column(name = "DiscountType", nullable = false, length = 20)
    open var discountType: String? = null

    @Column(name = "DiscountValue", nullable = false, precision = 18, scale = 2)
    open var discountValue: BigDecimal? = null

    @ColumnDefault("0")
    @Column(name = "MinOrderAmount", precision = 18, scale = 2)
    open var minOrderAmount: BigDecimal? = null

    @Column(name = "MaxDiscountAmount", precision = 18, scale = 2)
    open var maxDiscountAmount: BigDecimal? = null

    @Column(name = "UsageLimit")
    open var usageLimit: Int? = null

    @ColumnDefault("0")
    @Column(name = "UsedCount")
    open var usedCount: Int? = null

    @ColumnDefault("1")
    @Column(name = "IsActive")
    open var isActive: Boolean? = null

    @Column(name = "StartDate", nullable = false)
    open var startDate: Instant? = null

    @Column(name = "EndDate", nullable = false)
    open var endDate: Instant? = null

    @ColumnDefault("getdate()")
    @Column(name = "CreatedDate")
    open var createdDate: Instant? = null
}